File: app/app/src/main/java/com/example/ushopping/data/ProductListData.java
package com.example.ushopping.data;

import java.util.Date;
import java.util.UUID;

public class ProductListData {
    public UUID id;
    public String title;
    public Date createdAt;

    public ProductListData() {}

    public ProductListData(UUID id, String title, Date createdAt) {
        this.id = id;
        this.title = title;
        this.createdAt = createdAt;
    }
}

File: server/Data/ProductListData.cs
using System;
using uShopping.Models;
using System.Text.Json.Serialization;

namespace uShopping.Data {
  public class ProductListData {
    public Guid Id { get; set; }
    public string Title { get; set; }
    [JsonPropertyName("created_at")] public DateTime CreatedAt { get; set; }

    public ProductListData () {}
    public ProductListData (ProductList productList) {
        Id = productList.Id;
        Title = productList.Title;
        CreatedAt = productList.CreatedAt;
    }
  }
}